frlearn/uncategorised/fuzzy_quantifiers

```diff
@@ -0,0 +1,413 @@
+
+
+def QFM(gamma, semi_fuzzy_quantifier, set_v, fuzzy_sets):
+    """"
+    quantifier fuzzification mechanism
+    parameters
+    ----------
+    gamma: float
+               0 <= gamma <= 1
+        quantifier: function
+                    Define the semi-fuzzy quantifier function and use it as parameter
+        set_v: table (list)
+               The base set X on which the fuzzy set is defined
+        fuzzy_sets: table of table (list containing lists)
+                    Define the value of each fuzzy sets in the same list
+                    :param fuzzy_sets:
+                    :param set_v:
+                    :param gamma:
+                    :param semi_fuzzy_quantifier:
+    """
+    if bottom(gamma, semi_fuzzy_quantifier, set_v, fuzzy_sets) > 0.5:
+        return bottom(gamma, semi_fuzzy_quantifier, set_v, fuzzy_sets)
+    elif top(gamma, semi_fuzzy_quantifier, set_v, fuzzy_sets) < 0.5:
+        return top(gamma, semi_fuzzy_quantifier, set_v, fuzzy_sets)
+    else:
+        return 0.5
+
+
+def QFM_OWA(set_v, semi_fuzzy_quantifier, fuzzy_sets, n):
+    """""
+  The quantifier fuzzification mechanism OWA
+    Parameters:
+    ----------
+    set_v: table (list)
+           The base set X on which the fuzzy set is defined
+    quantifier: function
+                Define the semi-fuzzy quantifier function and use it as parameter
+    fuzzy_sets: table of table (list containing lists)
+                Define the value of each fuzzy sets in the same list
+    n: int
+       number of subdivisions of the interval [0,1]
+    """
+    h = 1 / n
+    result = 0.5 * ((top(0, semi_fuzzy_quantifier, set_v, fuzzy_sets) +
+                     bottom(0, semi_fuzzy_quantifier, set_v, fuzzy_sets)) / 2 +
+                    (top(1, semi_fuzzy_quantifier, set_v, fuzzy_sets) +
+                     bottom(1, semi_fuzzy_quantifier, set_v, fuzzy_sets)) / 2)
+    for i in range(1, n):
+        result += (top(i * h, semi_fuzzy_quantifier, set_v, fuzzy_sets) +
+                   bottom(i * h, semi_fuzzy_quantifier, set_v, fuzzy_sets)) / 2
+    result *= h
+    return result
+
+
+def Q_squared(rim, A, B):
+    return rim(len([el for el in A if el in B]) / len(A))
+
+
+def Q_arrow(rim, A, B, set_v):
+    return rim((len([el for el in set_v if el not in A]) + len([el for el in A if el in B])) / len(set_v))
+
+
+def W_implicator(set_v, f, rim, A, B=None):
+    """"
+    Evaluation of a binary sentence using a fuzzy quantifier based
+    on a Weighted Ordered Weighted Averaging (WOWA)
+    :parameter
+    set_v: table (list)
+           The base set X on which the fuzzy sets are defined
+    f: function
+       Define the Implicator
+    rim: function
+         define the rim quantifier
+    A, B: table (list)
+          List of the values of the fuzzy sets A and B
+    """
+    def mu(x):
+        return rim(sum([el for el in x if el in A]) / sum(A))
+    return choquet_integral(set_v, f, mu)
```

Comment:
This doesn't seem right, you define f as the implicator function (which is confusing, rename it) and then pass it directly to the Choquet integral function as the to be integrated function.

Reply:
Yes, in this function, f ix exactly I(A,B). I renamed it and made the requested changes